source/common/network/address_impl.cc

```
@@ -244,26 +244,44 @@ int Ipv6Instance::socket(SocketType type) const {
   return fd;
 }
 
-PipeInstance::PipeInstance(const sockaddr_un* address) : InstanceBase(Type::Pipe) {
+PipeInstance::PipeInstance(const sockaddr_un* address, socklen_t ss_len)
+    : InstanceBase(Type::Pipe) {
   if (address->sun_path[0] == '\0') {
-    throw EnvoyException("Abstract AF_UNIX sockets not supported.");
+    RELEASE_ASSERT(ss_len >= offsetof(struct sockaddr_un, sun_path) + 1);
+    abstract_namespace_ = true;
+    address_length_ = ss_len - offsetof(struct sockaddr_un, sun_path);
   }
   address_ = *address;
-  friendly_name_ = address_.sun_path;
+  friendly_name_ = abstract_namespace_
+                       ? "@" + std::string(address_.sun_path + 1, address_length_ - 1)
+                       : address_.sun_path;
 }
 
 PipeInstance::PipeInstance(const std::string& pipe_path) : InstanceBase(Type::Pipe) {
   memset(&address_, 0, sizeof(address_));
   address_.sun_family = AF_UNIX;
   StringUtil::strlcpy(&address_.sun_path[0], pipe_path.c_str(), sizeof(address_.sun_path));
   friendly_name_ = address_.sun_path;
+  if (address_.sun_path[0] == '@') {
+    abstract_namespace_ = true;
```

If this is not-linux, please throw an exception